staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -191,7 +191,16 @@ func (a *Authenticator) UpdateTransportConfig(c *transport.Config) error {
 		dial = (&net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}).DialContext
 	}
 	d := connrotation.NewDialer(dial)
+
+	// Before replace with new connrotation dialer, close all existing connections.
+	if a.onRotate != nil {
```

Comment:
all accesses to onRotate should be inside the lock

Reply:
Wrapped here. 
But for the other place you mentioned, it actually do not need to acquire the lock again due to the caller of `refreshCredsLocked()` are `getCreds()` and `maybeRefreshCreds()` which already acquire the lock.

https://github.com/kubernetes/kubernetes/blob/0599ca2bcfcae7d702f95284f3c2e2c2978c7772/staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go#L251-L252
https://github.com/kubernetes/kubernetes/blob/0599ca2bcfcae7d702f95284f3c2e2c2978c7772/staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go#L266-L267

Reply:
oh, missed that. thanks.